pennylane/data/__init__.py

```diff
@@ -13,22 +13,55 @@
 # limitations under the License.
 """The data subpackage provides functionality to access, store and manipulate quantum datasets.
 
+.. note::
+
+    For more details on using datasets, please see the
+    :doc:`quantum datasets quickstart guide </introduction/data>`.
+
+Overview
+--------
+
 Datasets are generally stored and accessed using the :class:`~pennylane.data.Dataset` class.
 Pre-computed datasets are available for download and can be accessed using the :func:`~pennylane.data.load` or
 :func:`~pennylane.data.load_interactive` functions.
 Additionally, users can easily create, write to disk, and read custom datasets using functions within the
 :class:`~pennylane.data.Dataset` class.
 
-.. currentmodule:: pennylane.data
 .. autosummary::
-   :toctree: api
+    :toctree: api
 
-Description
------------
+    attribute
+    field
+    Dataset
+    DatasetNotWriteableError
+    load
+    load_interactive
+    list_attributes
+    list_datasets
+
+In addition, various dataset types are provided
+
+.. autosummary::
+    :toctree: api
+
+    AttributeInfo
+    DatasetAttribute
+    DatasetArray
+    DatasetScalar
+    DatasetString
+    DatasetList
+    DatasetDict
+    DatasetOperator
+    DatasetNone
+    DatasetMolecule
+    DatasetSparseArray
+    DatasetJSON
+    DatsetTuple
```

Comment:
@timmysilv do these capture the ones you considered not user facing?

Reply:
yes! I'd also prefer if it's shoved off to the bottom and they're introduced more as something the user need not worry about, but the isolation looks good 👍